source/BadgeOverlaySupport.js

```diff
@@ -0,0 +1,220 @@
+	* the type of item. Like for Video/Audio it will show a default play icon and for Image
+	* there is nothing by default.
+	*
+	* The item has to define a
+	* [badgeType]{@link moon.BadgeOverlaySupport#badgeType} MIME type of this
+	* item. Possible values are 'video', 'audio' , 'image'. Defaulted as 'image'.
+	*
+	* ```javascript
+	* {name: 'list', kind: 'moon.DataList', components: [
+	*	{mixins: ['moon.BadgeOverlaySupport'], badgeType: 'video',
+	*		kind: 'moon.ImageItem', bindings: [
+	*			{from: '.model.title', to: '.label'},
+	*			{from: '.model.description', to: '.text'},
+	*			{from: '.model.coverSource', to: '.source'}
+	*		]
+	*	}
+	* ]}
+	* ```
+	*
+	* The item may define a
+	* [badgeSrc]{@link moon.BadgeOverlaySupport#badgeSrc} URL to
+	* override the default icon.
+	*
+	* ```javascript
+	* {name: 'list', kind: 'moon.DataList', components: [
+	*	{mixins: ['moon.BadgeOverlaySupport'], badgeType: 'video', badgeSrc: 'assets/my-icon.png',
+	*		kind: 'moon.ImageItem', bindings: [
+	*			{from: '.model.title', to: '.label'},
+	*			{from: '.model.description', to: '.text'},
+	*			{from: '.model.coverSource', to: '.source'}
+	*		]
+	*	}
+	* ]}
+	* ```
+	*
+	* The item may define a
+	* [position]{@link moon.BadgeOverlaySupport#position} to
+	* specify position of badge. Possible values are 'top' or 'bottom'. Defaulted as 'bottom'.
+	* ```javascript
+	* {name: 'list', kind: 'moon.DataList', components: [
+	*	{mixins: ['moon.BadgeOverlaySupport'], badgeType: 'video', position: 'top',
+	*		kind: 'moon.ImageItem', bindings: [
+	*			{from: '.model.title', to: '.label'},
+	*			{from: '.model.description', to: '.text'},
+	*			{from: '.model.coverSource', to: '.source'}
+	*		]
+	*	}
+	* ]}
+	*
+	* @mixin moon.BadgeOverlaySupport
+	* @protected
+	*/
+	moon.BadgeOverlaySupport = {
+
+		/**
+		* @private
+		*/
+		name: 'moon.BadgeOverlaySupport',
+
+		/**
+		* @private
+		*/
+		classes: 'moon-badge-overlay-support',
+
+		/**
+		* Identifies the MIME type of badge.
+		* Possible values are "image", "audio" or "video".
+		* Defaulted as 'image'.
+		*
+		* @name moon.BadgeOverlaySupport#badgeType
+		* @type {String}
+		* @default undefined
+		* @public
+		*/
+		badgeType: null,
+
+		/**
+		* Font icon to be used in place of default icon.
+		*
+		* @name moon.BadgeOverlaySupport#badgeIcon
+		* @type {String}
+		* @default undefined
+		* @public
+		*/
+		badgeIcon: null,
+
+		/**
+		* Url for icon to be used in place of default icon.
+		*
+		* @name moon.BadgeOverlaySupport#badgeSrc
+		* @type {String}
+		* @default undefined
+		* @public
+		*/
+		badgeSrc: null,
+
+		/**
+		* Sepcifies the alignment position of badge.
+		* Possible values are 'top' or 'bottom'.
+		* Defaulted as 'bottom'.
+		*
+		* @name moon.BadgeOverlaySupport#position
+		* @type {String}
+		* @default undefined
+		* @public
+		*/
+		position: null,
+
+		/**
+		* @private
+		*/
+		events: {
+
+			/**
+			* {@link moon.BadgeOverlaySupport#onBadgeTap}
```

Comment:
These don't link anywhere because the events haven't been documented correctly. See https://github.com/enyojs/moonstone/blob/master/source/Calendar.js#L2 for an example of documenting an event.